File: SDKs/amazonaws.com/waf/2015-08-24/go/pkg/models/operations/createsqlinjectionmatchset.go
package operations

import (
"openapi/pkg/models/shared")


type CreateSQLInjectionMatchSetXAmzTargetEnum string

const (
    CreateSQLInjectionMatchSetXAmzTargetEnumAwswaf20150824CreateSQLInjectionMatchSet CreateSQLInjectionMatchSetXAmzTargetEnum = "AWSWAF_20150824.CreateSqlInjectionMatchSet"
)


type CreateSQLInjectionMatchSetHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    XAmzTarget CreateSQLInjectionMatchSetXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
    
}

type CreateSQLInjectionMatchSetRequest struct {
    Headers CreateSQLInjectionMatchSetHeaders 
    Request shared.CreateSQLInjectionMatchSetRequest `request:"mediaType=application/json"`
    
}

type CreateSQLInjectionMatchSetResponse struct {
    ContentType string 
    CreateSQLInjectionMatchSetResponse *shared.CreateSQLInjectionMatchSetResponse 
    StatusCode int64 
    WafDisallowedNameException *interface{} 
    WafInternalErrorException *interface{} 
    WafInvalidAccountException *interface{} 
    WafInvalidParameterException *interface{} 
    WafLimitsExceededException *interface{} 
    WafStaleDataException *interface{} 
    
}

File: SDKs/amazonaws.com/waf-regional/2016-11-28/typescript/src/sdk/models/operations/createsqlinjectionmatchset.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";

export enum CreateSqlInjectionMatchSetXAmzTargetEnum {
    AwswafRegional20161128CreateSqlInjectionMatchSet = "AWSWAF_Regional_20161128.CreateSqlInjectionMatchSet"
}


export class CreateSqlInjectionMatchSetHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: CreateSqlInjectionMatchSetXAmzTargetEnum;
}


export class CreateSqlInjectionMatchSetRequest extends SpeakeasyBase {
  @Metadata()
  headers: CreateSqlInjectionMatchSetHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.CreateSqlInjectionMatchSetRequest;
}


export class CreateSqlInjectionMatchSetResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  createSqlInjectionMatchSetResponse?: shared.CreateSqlInjectionMatchSetResponse;

  @Metadata()
  statusCode: number;

  @Metadata()
  wafDisallowedNameException?: any;

  @Metadata()
  wafInternalErrorException?: any;

  @Metadata()
  wafInvalidAccountException?: any;

  @Metadata()
  wafInvalidParameterException?: any;

  @Metadata()
  wafLimitsExceededException?: any;

  @Metadata()
  wafStaleDataException?: any;
}
